File: ruby/easy/maximum_distance_in_arrays.rb
# @param {Integer[][]} arrays
# @return {Integer}
def max_distance(arrays)
    min = arrays[0].first
    max = arrays[0].last
    max_distance = 0
    for i in 1..arrays.length-1
        a_min = arrays[i].first
        a_max = arrays[i].last
        max_distance = [max_distance, (a_max - min).abs, (max - a_min).abs].max
        min = [a_min, min].min
        max = [a_max, max].max
    end
    return max_distance
end

File: javascript/easy/maximumDistanceInArrays.js
/**
 * @param {number[][]} arrays
 * @return {number}
 */
var maxDistance = function(arrays) {
    let min = arrays[0][0], max = arrays[0][arrays[0].length-1], maxDistance = 0;
    for(let i = 1; i < arrays.length; i++) {
        let aMin = arrays[i][0], aMax = arrays[i][arrays[i].length-1];
        maxDistance = Math.max(maxDistance, Math.abs(max - aMin), Math.abs(aMax - min));
        min = Math.min(min, aMin);
        max = Math.max(max, aMax);
    }
    return maxDistance;
};
